### PR TITLE
Exercise folder names: Cherry pick folder renames from main to solutions branch

### PR DESCRIPTION
**Pull request by richfromm**
## Because

After solving the exercises, it's useful to compare your results to the solutions. This is significantly easier if the file names are the same on the solutions branch, since you can then use git operations to do so. For example, with this change, I can within emacs access the version of the same file on the solutions branch with "C-x v ~" (Show Other Version).

From the final item in the checkbox list of Pull Request Requirements, it sounds like maybe this should have happened back when the folders were renamed on the main branch, in August 2021?

## This PR

* Cherry pick the following commits from main (listed in `git log` order, so reverse time order)
```
3236ae156ccc536c34435b134e6232d1f930bc3c Revert edits
61f38bf60cedaf210ddd067c624a086e0f2c7a71 Rename folders with numbers and underscores
55cfb173d8da273146a5c598cb8ea3e9760831fa Order exercise folders
```

## Issue

*(no issue)*

## Additional Information

It took me a few tries to get this right, because I didn't originally realize that there were 3 original commits, not 2, and those were kind of confusing and including reverting some unintentional edits.

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `01_helloWorld: Update test cases`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [ ] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If this PR includes changes that needs to be updated on the `solutions` branch, I have created another PR (and linked it to this PR).

### CODE REVIEW

File: 03_reverseString/reverseString.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const reverseString = function(str) {
```
Comment (richfromm) [inline 03_reverseString/reverseString.js:1]: No, this is wrong. I think something got messed up, including some of my work from doing this. Investigating further, may delete this all and start over.

Comment (richfromm) [reply 03_reverseString/reverseString.js:1]: Okay, I now understand this better. The original renaming took place over 3 commits, not 2:
* https://github.com/TheOdinProject/javascript-exercises/commit/55cfb173d8da273146a5c598cb8ea3e9760831fa
* https://github.com/TheOdinProject/javascript-exercises/commit/61f38bf60cedaf210ddd067c624a086e0f2c7a71
* https://github.com/TheOdinProject/javascript-exercises/commit/3236ae156ccc536c34435b134e6232d1f930bc3c

And the 2nd of those commits did what I thought I had done, which was it inadvertently checked in a solution to one of the exercises. The third commit attempted to undo that.

But applying all of that now on the solutions branch, the end result is that the actual solution disappeared. I will fix that in another commit.

File: 03_reverseString/reverseString.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 const reverseString = function(string) {
- return string.split('').reverse().join('');
+  return string.split('').reverse().join('');
```
Comment (richfromm) [inline 03_reverseString/reverseString.js:2]: Finally, after a bunch of confusion, this is logically returned to the way it was.

I do see a whitespace change crept in, but I think it's for the better (and consistent with other cases), so I'll leave it.

**Comment by richfromm** [comment]
I'm hesitant to put it here, b/c I think it's convoluting separate issues and should perhaps belong in a separate PR, but part of my difficulty in applying the cherry picks was a previous inadvertent substitution that changed some of the `test.skip()` calls to `test()`.

But shouldn't the solutions branch have that? Should perhaps all of the tests be active, with none skipped, so that someone can run all of the tests on the solutions branch and verify that they indeed pass?

**Comment by richfromm** [comment]
Attn: @remtaine , as you were the original author of the commits I cherry-picked.

**Comment by Asartea** [comment]
See https://github.com/TheOdinProject/javascript-exercises/pull/303 (attack @thatblindgeye )

**Comment by richfromm** [comment]
I agree that https://github.com/TheOdinProject/javascript-exercises/pull/234 is probably a better long term solution, to refactor the main branch to have solutions in individual `solution` subdirs, rather than have a separate `solutions` branch.

But that looks like it's a bit stalled, it's nearly a year old. If that's not going to merge any time soon, might this still be worthwhile as a short term solution? Are there any downsides to merging this now?

**Comment by remtaine** [comment]
Thanks for tagging me! Sorry, it's been a while since I've done the Odin Project. From what I understand, there was a problem with renaming the folders for the main branch but not the solutions branch? If so, I can rename it so it'd be easier to match the exercises and the solutions.

With that said, I agree that having solutions in a sub-directory through #234 would have been better as well. The reason why I didn't do the previously suggested merge was because I didn't notice the solutions branch back then as a beginner.

**Comment by Asartea** [comment]
> From what I understand, there was a problem with renaming the folders for the main branch but not the solutions branch?

Yeah, mostly because it means that due to the way Git (and GitHub) work you can only switch branches in the root directory, because switching lower down will simply give you page not found errors

**Comment by remtaine** [comment]
Okay, this is noted. So, either:

1) we go with #234, or 
2) I can do move the solutions branch into a subdirectory of the main branch, as well as rename the folders 

Just tell me if Option 2 is preferred. If so, what do you think is better: renaming the subdirectories in solutions folder to have the numbers or removing the numbers from the main branch?

**Comment by richfromm** [comment]
I agree that https://github.com/TheOdinProject/javascript-exercises/pull/234 is a better solution. My point was just that if it's going to be delayed indefinitely, maybe this had value in the interim.

But perhaps it just kind of got forgotten about and has now been revived and may get done? I see that @thatblindgeye changed its Status from Do Not Review to Needs Review 2 weeks ago.
 
Also, I feel strongly that the number prefixes should stay, since it means the order of a directory listing corresponds to the order in which the exercises should be done.

**Comment by thatblindgeye** [comment]
For now will be closing this since this would be covered in the above linked PR.

**Pull request closed without merging**